### ISSUE TITLE
Bug with windows on top of each other

### ISSUE THREAD

**Issue by ImperialUser** [Heishe/idleraiders-second-run#413]
The game doesn't recognize if something is on top of something else. (firefox browser, I haven't tested it in others) A couple examples:

If you have the crafting window open in town with the close (X) button on top of another building it will both close your window and open the new window for the building it was over.  If it's over the crafting building it essentially doesn't close at all because it closes and then opens again right away.

If you have one window on top of another it can cause them both do weird things, or just not work at all.

If you have a piece of equipment selected and choose to sort items by useable, this puts your mouse over the trash button and you delete the piece of equipment you had selected as well.

**Comment by TravisGesslein** [comment]
True, there are some bugs left like this. We have a kind of system where we say 'this piece of the UI belongs to window x', where a window is something like the inventory screen or the equipment screen. Since we introduced that windowing system a long time after we implemented most of the UI in the alpha, there are still some places left where elements aren't assigned to the right window, so the clicking-elements-beneath stuff that you mention happens sometimes. We'll have to find each one of those that still happens and then fix those specifically.

**Comment by Profputricide** [comment]
The snapshotting for full team also has this issue. You can't open the snapshot window, if the close button of the snapshot window is on top of the icon in equipment window.
![](http://puu.sh/sOYYw/eecb35d05e.png)

**Comment by Profputricide** [comment]
Mercenary window is also clicked through
